src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -516,15 +516,22 @@ private static string GetNegotiatePayload(string connectionId, HttpSocketOptions
             return connection;
         }
 
-        private async Task<DefaultConnectionContext> GetOrCreateConnectionAsync(HttpContext context)
+        private DefaultConnectionContext CreateConnectionInternal(HttpSocketOptions options)
+        {
+            var transportPipeOptions = new PipeOptions(pauseWriterThreshold: options.TransportMaxBufferSize, resumeWriterThreshold: options.TransportMaxBufferSize / 2);
+            var appPipeOptions = new PipeOptions(pauseWriterThreshold: options.ApplicationMaxBufferSize, resumeWriterThreshold: options.ApplicationMaxBufferSize / 2);
+             return _manager.CreateConnection(transportPipeOptions, appPipeOptions);
```

Comment:
nit: formatting is off here. Do you have the same settings as @anurse ? :trollface: